tests/conftest.py

```diff
@@ -446,14 +446,22 @@ def reset_all_objects():
     unknown_parameters.clear()
 
 
+@pytest.fixture(scope="session", autouse=True)
+def init_cwd():
+    """
+    Fixture returning the cwd at the beginning of the session.
+    """
+    yield os.getcwd()
```

Comment:
I think, that a very long time ago, the os.getcwd() was captured when conftest was loaded (just a variable in the module), that is guaranteed to be very early, but I don't know why this was changed (too early?)

Reply:
Nice idea. I tested it and it seems to works fine.